File: tracking-program/8087/api/models/Gateway.js
/**
 * Gateway.js
 *
 * @description :: A model definition.  Represents a database table/collection/etc.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {

    // Serial number of the network device (e.g. Pi-board) being used as a gateway.
    deviceId: {
      type: 'string',
      required: true,
      custom: function(val) {
        return val.trim().toUpperCase().match(/^([A-F0-9]{2}:){5}[A-F0-9]{2}$/);
      }
    },

    // MAC address of the this gateway's mesh node (e.g. the Pi-board's wireless controller MAC address).
    meshId: {
      type: 'string'
    },

    meshIp: {
      type: 'string',
      allowNull: true,
    },

    // Nickname for this gateway, e.g. "BLDG #1 Gateway".
    name: {
      type: 'string',
      required: true
    },

    // The current (acknowledged) software version on the gatewaty.
    softwareVersion: {
      type: 'string'
    },

    // The project that this gateway is attached to.
    project: {
      model: 'project',
      required: true
    },

    // Time when the gateway last communicated with the server.
    lastCommunicatedAt: {
      type: 'number'
    },

    // Whether or not this gateway has been deleted.
    isDeleted: {
      type: 'boolean',
      defaultsTo: false
    },

    tests: {
      collection: 'test',
      via: 'gateways'
    }


  },

  beforeCreate: trimDeviceId,
  beforeUpdate: trimDeviceId

};

function trimDeviceId(vals, cb) {
  if (vals.deviceId) {
    vals.deviceId = vals.deviceId.trim().toUpperCase();
  }
  return cb(null, vals);
}

File: tracking-program/8087/flask_app/app/models/gateway.py
"""Gateway model - api/models/Gateway.js"""
from app.extensions import db
from app.models.base import BaseModel


class Gateway(BaseModel):
    __tablename__ = "gateway"

    deviceId = db.Column(db.String(255), nullable=False)
    meshId = db.Column(db.String(255))
    meshIp = db.Column(db.String(255), nullable=True)
    name = db.Column(db.String(255), nullable=False)
    softwareVersion = db.Column(db.String(255))
    project = db.Column(db.Integer, db.ForeignKey("project.id"), nullable=False)
    lastCommunicatedAt = db.Column(db.Float)
    isDeleted = db.Column(db.Boolean, default=False)
